src/session/session_api.c

```diff
@@ -700,6 +700,10 @@ __session_rebalance(WT_SESSION *wt_session, const char *uri, const char *config)
 
 	SESSION_API_CALL(session, rebalance, config, cfg);
 
+	/* In-memory doesn't support rebalance operations. */
+	if (F_ISSET(S2C(session), WT_CONN_IN_MEMORY))
+		goto err;
+
```

Comment:
Should this go down into `wt_schema_worker` instead of faulting it into each caller?  There are several callers of it in `session_api.c` that require exclusive access, rebalance, salvage, upgrade, verify and alter (on my branch).  Salvage has an error check.  The others do not.  Isn't it also needed on upgrade and verify? It should also be documented in the docs that some functions are invalid with in-memory in docs/in_memory.dox.